src/workerd/io/worker.c++

```diff
@@ -65,6 +65,9 @@ namespace workerd {
 
 namespace {
 
+constexpr int32_t kMinSmi = 0xc0000000;
+constexpr int32_t kMaxSmi = -(kMinSmi + 1);
```

Comment:
It seems quite plausible that this will wrap around in high-traffic workers over a week or two.

But I guess we are OK with a very rare false negative? Should be clear about that in comments.

Reply:
Yeah honestly I'm not sure what is best here. I think we should try to avoid the false negatives if we can do so reasonably but Smi is obviously going to be the most performant choice here. tl;dr I don't know